In [2]:
import numpy as np
import pandas as pd
from matplotlib import pyplot as plt
import math
from scipy import stats as stats


### Задача № 1

Даны две независимые выборки. Не соблюдается условие нормальности
x1 380,420, 290
y1 140,360,200,900
Сделайте вывод по результатам, полученным с помощью функции, имеются ли статистические различия между группами?


##### Решение

Так как не соблюдается условие нормальности,   две группы и две независимые выборки, то используем U-критерий Манна-Уитни

In [3]:
x1 = ([380, 420,290])
x2 = ([140, 360, 200, 900])
stats. mannwhitneyu(x1,x2)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

pvalue = 0.629 > alpha = 0.005, не отвергаем нулевую гипотезу на уровне значимости alpha = 0.005,  делаем вывод, что между группами нет  статистически знначимыч различий. 

###  Задача № 2

Исследовалось влияние препарата на уровень давления пациентов. Сначала
измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть
ли статистически значимые различия между измерениями давления? В выборках не соблюдается условие нормальности.
1е измерение до приема препарата: 150, 160, 165, 145, 155
2е измерение через 10 минут: 140, 155, 150, 130, 135
3е измерение через 30 минут: 130, 130, 120, 130, 125

#### Решение

In [4]:
before = np.array([150, 160 ,165, 145, 155])
after_10m = np.array([140, 155 ,150, 130, 135])
after_30m = np.array([130, 130 ,120, 130, 125])

stats.friedmanchisquare(before, after_10m, after_30m)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

pvalue = 0.008 > alpha = 0.005, не отвергаем нулевую гипотезу на уровне значимости alpha = 0.005,  делаем вывод, что между группами нет  статистически значимых различий. 

### Задача № 3

Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было. Есть
ли статистически значимые различия между измерениями давления?

#### Решение

При условии отсутствия третьего измерения мы имеем две засимые выборки и воспользуемся критерием Уилкоксона

In [5]:
before = np.array([150, 160 ,165, 145, 155])
after_10m = np.array([140, 155 ,150, 130, 135])

stats.wilcoxon(before, after_10m)


WilcoxonResult(statistic=0.0, pvalue=0.0625)

pvalue = 0.006 > alpha = 0.005, не отвергаем нулевую гипотезу на уровне значимости alpha = 0.005,  делаем вывод, сто между группами нет  статистически знначимых различий. 

### Задача № 4

Даны 3 группы учеников плавания. Не соблюдается условие нормальности.
В 1 группе время на дистанцию 50 м составляют:
56, 60, 62, 55, 71, 67, 59, 58, 64, 67
Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53
Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54
Есть
ли статистически значимые различия между группами?

#### Решение

Так как мы сравниваем болле 2-х независимых выборок,  также не соблюдается условие нормальности, то применим критерий Крускала – Уоллиса 

In [6]:
gr_1 = np.array ([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
gr_2 = np.array ([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
gr_3 = np.array ([57, 67, 49, 48, 47, 55, 66, 51, 54])

stats.kruskal(gr_1, gr_2, gr_3)

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

pvalue = 0.065 > alpha = 0.005, не отвергаем нулевую гипотезу на уровне значимости alpha = 0.005, делаем вывод, что между группами нет статистически знначимых различий

### Задача № 5

Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить
данную гипотезу, если известно, что размеры изделий подчинены нормальному закону
распределения. Объем выборки 10, уровень статистической значимости 5%
2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

Для проверки гипотезы о среднем арифметическом при нормальном распределении используется t-критерий Стьюдента.

In [39]:
part = np.array ([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])

In [40]:
alpha = 0.005
n = 10
mu = 2.5

In [41]:
part_mean =  part.sum() / part.size  # среднее выборочное
part_mean

2.5279999999999996

In [42]:
sigma = (np.sum((part - part_mean)**2) / (part.size -1 ))**0.5  #стандартное отклонение
sigma

0.1572542173961923

In [43]:
T1 = stats.t.ppf(alpha/2, df=n-1 )
T2 = stats.t.ppf(1 - alpha/2, df=n-1 )
T1, T2

(-3.689662392304235, 3.6896623923042484)

In [44]:
T_emp = (part_mean - mu) / (sigma /np.sqrt(n))
T_emp 

0.5630613661802959

Критическая область ( - inf, -3,69) ( 3,69, inf ) T_emp не принадлежит критической области, нулевая гипотеза не отвергается на уровне значимости 5%. Утверждение, что партия изготавливается со средним арифметическим 2,5 верно.